sdk/messaging/azeventhubs/event_data.go

```diff
@@ -127,35 +109,60 @@ func newReceivedEventData(amqpMsg *amqp.Message) *ReceivedEventData {
 		}
 
 		re.ContentType = amqpMsg.Properties.ContentType
+		re.CorrelationID = amqpMsg.Properties.CorrelationID
 	}
 
 	if amqpMsg.ApplicationProperties != nil {
-		re.ApplicationProperties = make(map[string]any, len(amqpMsg.ApplicationProperties))
+		re.Properties = make(map[string]any, len(amqpMsg.ApplicationProperties))
 		for key, value := range amqpMsg.ApplicationProperties {
-			re.ApplicationProperties[key] = value
+			re.Properties[key] = value
 		}
 	}
 
-	if amqpMsg.Annotations != nil {
-		if sequenceNumber, ok := amqpMsg.Annotations[sequenceNumberAnnotation]; ok {
-			re.SequenceNumber = sequenceNumber.(int64)
-		}
+	updateFromAMQPAnnotations(amqpMsg, re)
+	return re
+}
 
-		if partitionKey, ok := amqpMsg.Annotations[partitionKeyAnnotation]; ok {
-			re.PartitionKey = to.Ptr(partitionKey.(string))
-		}
+// the "SystemProperties" in an EventData are any annotations that are
+// NOT available at the top level as normal fields. So excluing  sequence
+// number, offset, enqueued time, and  partition key.
+func updateFromAMQPAnnotations(src *amqp.Message, dest *ReceivedEventData) {
+	if src.Annotations == nil {
+		return
+	}
+
+	for kAny, v := range src.Annotations {
+		keyStr, keyIsString := kAny.(string)
 
-		if enqueuedTime, ok := amqpMsg.Annotations[enqueuedTimeAnnotation]; ok {
-			t := enqueuedTime.(time.Time)
-			re.EnqueuedTime = &t
+		if !keyIsString {
+			continue
 		}
 
-		if offset, ok := amqpMsg.ApplicationProperties[offsetNumberAnnotation]; ok {
-			if offsetInt64, ok := offset.(int64); ok {
-				re.Offset = to.Ptr(int64(offsetInt64))
+		switch keyStr {
+		case sequenceNumberAnnotation:
+			if asInt64, ok := v.(int64); ok {
+				dest.SequenceNumber = asInt64
+			}
+		case partitionKeyAnnotation:
+			if asString, ok := v.(string); ok {
+				dest.PartitionKey = to.Ptr(asString)
+			}
+		case enqueuedTimeAnnotation:
+			if asTime, ok := v.(time.Time); ok {
+				dest.EnqueuedTime = &asTime
+			}
+		case offsetNumberAnnotation:
+			if offsetStr, ok := v.(string); ok {
+				if offset, err := strconv.ParseInt(offsetStr, 10, 64); err == nil {
```

Comment:
What happens if `err != nil`?

Reply:
Same here. If they send us invalid data we just leave Offset uninitialized (ie, 0)

Reply:
We return an error now. It'll abort the Receive operation but this isn't as catastrophic as it would be for service bus.